File: Javascript/fizzbuzz.js
// Run:
//
//   $ node fizzbuzz.js
//

function fizzbuzz(limit, config) {
  return [config[0].message]
}

const limit = 10
const config = [
  { "modulo": 3, "message": "Fizz" },
  { "modulo": 5, "message": "Buzz" },
]

console.log(`limit ${limit}`)
for (var configEntry of config) {
  console.log(`config (${configEntry.modulo}, ${configEntry.message})`)
}

for (var result of fizzbuzz(limit, config)) {
  console.log(result)
}

File: Ruby/fizzbuzz.rb
# Run:
#
#   $ ruby fizzbuzz.rb
#

ConfigEntry = Struct.new(:modulo, :message)

def fizzbuzz(limit, config)
  [config.first.message]
end

limit = 10
config = [
  ConfigEntry.new(3, "Fizz"),
  ConfigEntry.new(5, "Buzz"),
]

puts "limit #{limit}"
config.each do |config_entry|
  puts "config (#{config_entry.modulo}, #{config_entry.message})"
end

fizzbuzz(limit, config).each do |result|
  puts result
end
